lib/rules/prefer-regex-literals.js

```diff
@@ -306,10 +354,61 @@ module.exports = {
 
                 for (const { node } of tracker.iterateGlobalReferences(traceMap)) {
                     if (disallowRedundantWrapping && isUnnecessarilyWrappedRegexLiteral(node)) {
+                        const regexNode = node.arguments[0];
+
                         if (node.arguments.length === 2) {
-                            context.report({ node, messageId: "unexpectedRedundantRegExpWithFlags" });
+                            const outputs = [];
+
+                            if (sourceCode.getCommentsInside(node).length <= 0) {
+                                const literalFlags = regexNode.regex.flags || "";
+                                const argFlags = getStringValue(node.arguments[1]) || "";
+
+                                if (isValidRegexForEcmaVersion(regexNode.regex.pattern, argFlags)) {
+                                    outputs.push(`/${regexNode.regex.pattern}/${argFlags}`);
+                                }
+
+                                if (literalFlags && argFlags) {
+                                    const mergedFlags = mergeRegexFlags(literalFlags, argFlags);
+
+                                    if (
+                                        !isFlagsEqual(mergedFlags, argFlags) &&
+                                        isValidRegexForEcmaVersion(regexNode.regex.pattern, mergedFlags)
+                                    ) {
+                                        outputs.push(`/${regexNode.regex.pattern}/${mergedFlags}`);
+                                    }
+                                }
+                            }
+
+                            context.report({
+                                node,
+                                messageId: "unexpectedRedundantRegExpWithFlags",
+                                suggest: outputs.map(output => ({
+                                    messageId: "replaceWithLiteralAndFlags",
+                                    fix(fixer) {
+                                        return fixer.replaceText(node, output);
+                                    }
+                                }))
+                            });
                         } else {
-                            context.report({ node, messageId: "unexpectedRedundantRegExp" });
+                            const outputs = [];
+
+                            if (
+                                sourceCode.getCommentsInside(node).length <= 0 &&
```

Comment:
I don't think this can be less than zero. :)

```suggestion
                                sourceCode.getCommentsInside(node).length === 0 &&
```